flair/models/biomedical_entity_linking.py

```diff
@@ -0,0 +1,1187 @@
+
+        return hybrid_scores, hybrid_ids
+
+    def search(self, entity_mentions: List[str], top_k: int) -> List[List[Tuple[str, str, float]]]:
+        """
+        Returns the top-k entity / concept identifiers for the each entity mention.
+
+        :param entity_mentions: Entity mentions
+        :param top_k: Number of best-matching entities from the knowledge base to return
+        :result: list of tuples in the form: (entity / concept name, concept ids, similarity score).
+        """
+
+        scores, ids = self.search_dense(entity_mentions=entity_mentions, top_k=top_k)
+
+        if self.hybrid_search:
+
+            sparse_scores, sparse_ids = self.search_sparse(entity_mentions=entity_mentions, top_k=top_k)
+
+            scores, ids = self.combine_dense_and_sparse_results(
+                dense_ids=ids,
+                dense_scores=scores,
+                sparse_scores=sparse_scores,
+                sparse_ids=sparse_ids,
+                top_k=top_k,
+            )
+
+        return [
+            [tuple(self.dictionary_data[i]) + (score,) for i, score in zip(mention_ids, mention_scores)]
+            for mention_ids, mention_scores in zip(ids, scores)
+        ]
+
+
+class BiomedicalEntityLinker:
+    """Entity linking model for the biomedical domain"""
+
+    def __init__(
+        self,
+        candidate_generator: AbstractCandidateGenerator,
+        preprocessor: AbstractEntityPreprocessor,
+        entity_type: str,
+    ):
+        self.preprocessor = preprocessor
+        self.candidate_generator = candidate_generator
+        self.entity_type = entity_type
+        self.annotation_layers = [ENTITY_TYPE_TO_ANNOTATION_LAYER.get(self.entity_type, "ner")]
+
+    def build_candidate(self, candidate: Tuple[str, str, float]):
+        """Get nice container with all info about entity linking candidate"""
+
+        concept_name = candidate[0]
+        concept_id = candidate[1]
+        score = candidate[2]
+        database_name = self.candidate_generator.dictionary.database_name
+
+        if "|" in concept_id:
+            labels = concept_id.split("|")
+            concept_id = labels[0]
+            additional_labels = labels[1:]
+        else:
+            additional_labels = None
+
+        return EntityLinkingCandidate(
+            concept_id=concept_id,
+            concept_name=concept_name,
+            score=score,
+            additional_ids=additional_labels,
+            database_name=database_name,
+        )
+
+    def extract_mentions(
+        self,
+        sentences: List[Sentence],
+        annotation_layers: Optional[List[str]] = None,
+    ) -> Tuple[List[int], List[Span], List[str]]:
+        """Unpack all mentions in sentences for batch search."""
+
+        source = []
+        data_points = []
+        mentions = []
+        mention_annotationa_layers = []
+
+        # get all valid annotation layers: pre-determined and user input
+        annotation_layers = (
```

Comment:
Hey @mariosaenger I have addressed all your comments. 

Honestly I am still a bit confused by  the `annotation_layer` business. Can you take a look at this line and tell me if it's ok?

Reply:
Mhh I would rather opt to apply the default annotation layers only iff the user doesn't provide a value for the annotation layers parameter. Otherwise the user has no full control over the behaviour (i.e. he/she can't get rid of the default annotation layers).